duckduckgo/Android
# Create internal flavor

Task/Issue URL: https://app.asana.com/0/414730916066338/1200572408644661/f
Tech Design URL: 
CC: 

**Description**:
We want to add an `internal` production flavor that will be used to release the app for internal users. The main use-case for the this `internal` flavor is to host some features that serve no purpose in production, eg. debug/dev panel

While doing that, I have also cleaned-up a bit the `play` and `fdroid` flavors, more details below.

**Changes in this PR**
* add the `internal` flavor to the `app/build.gradle`
* `internal` flavor only has `release` version, similar to what we have for `fdroid`
 * If we see this causes friction while developing we will enable the `internalDebug` as well
* `internal` flavor does not contain anything yet
* Tidy up for `play` and `fdroid` flavors
 * both flavors were providing the same `AppInstallationReferrerParser` and `AppReferrerDataStore` implementations, so I moved those providers out of the `StoreReferralModule`
 * as it was, every flavor had to provide its own `StoreReferralModule` and `HttpsPersisterModule` modules, even if the flavor was not concern with those. This would have been the case for the `internal` module
 * what we do now is to provide default implementations in `main` and any flavor that wants to override the default behavior can do so using DI. This is why I ahve:
  * moved the F-Droid `HttpsPersisterModule` and the `StoreReferralModule` into `main` (this will serve as default impls)
  * removed the F-Droid `HttpsPersisterModule` and the `StoreReferralModule` to the main app
  * `play` flavor will override those to DI modules by its own `PlayHttpsPersisterModule` and `PlayStoreReferralModule`
  * the reason we used the F-Droid as default implementation is because F-Droid APK can't contain the `bloom` binary file nor the `com.android.installreferrer` library


**Steps to test this PR**:

Test Play flavor
1. in this branch, optionally place some timber statements `PlayStoreAppReferrerStateListener` and `PlayHttpsEmbeddedDataPersister`
1. install the playDebug from this branch
1. launch the app
1. verify the `PlayHttpsEmbeddedDataPersister` and `PlayStoreAppReferrerStateListener` are called respectively

Test Fdroid/Internal flabors
1. install the fdroid/internal flavors
1. launch the app
1. verify `EmptyReferrerStateListener` and `EmptyHttpsEmbeddedDataPersister` are called instead


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
